cmake/modules/FindDate.cmake

```
@@ -0,0 +1,128 @@
+                      CONFIGURE_COMMAND ""
+                      BUILD_COMMAND ""
+                      INSTALL_COMMAND ${CMAKE_COMMAND} -E copy_directory <SOURCE_DIR> <INSTALL_DIR>
+                      BUILD_IN_SOURCE ON)
+
+  list(APPEND GLOBAL_TARGET_DEPS tzdata)
+endif()
+
+if(ENABLE_INTERNAL_DATE)
+  include(ExternalProject)
+
+  # Build internal date library
+
+  # Allow user to override the download URL with a local tarball
+  # needed for offline build envs
+
+  if(DATE_URL)
+    get_filename_component(DATE_URL "${DATE_URL}" ABSOLUTE)
+  else()
+    set(DATE_URL https://github.com/HowardHinnant/date/archive/refs/tags/v${DATE_VERSION}.tar.gz)
+  endif()
+
+  if(VERBOSE)
+    message(STATUS "DATE_URL: ${DATE_URL}")
+  endif()
+
+  set(DATE_LIBRARY ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/lib/libdate-tz.a)
+  set(DATE_INCLUDE_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/include)
+
+  set(DATE_CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+                      -DCMAKE_CXX_STANDARD=17
+                      -DUSE_SYSTEM_TZ_DB=OFF
+                      -DMANUAL_TZ_DB=ON
+                      -DUSE_TZ_DB_IN_DOT=OFF
+                      -DBUILD_SHARED_LIBS=OFF
+                      -DBUILD_TZ_LIB=ON
+                      -DCMAKE_INSTALL_LIBDIR=lib
+                      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE})
+
+  externalproject_add(Date
+                      URL ${DATE_URL}
+                      DOWNLOAD_NAME date-${DATE_VERSION}.tar.gz
+                      DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/download
+                      PREFIX ${CORE_BUILD_DIR}/date
+                      INSTALL_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}
+                      CMAKE_ARGS ${DATE_CMAKE_ARGS}
+                      BUILD_BYPRODUCTS ${DATE_LIBRARY})
+
+  externalproject_add_step(Date install_headers
+                          COMMAND ${CMAKE_COMMAND} -E copy include/date/iso_week.h ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/include/date/iso_week.h
+                          WORKING_DIRECTORY <SOURCE_DIR>
+                          DEPENDEES install)
+
+  set_target_properties(Date PROPERTIES FOLDER "External Projects")
+  list(APPEND GLOBAL_TARGET_DEPS Date)
+else()
+  find_path(DATE_INCLUDE_DIR NAMES date.h)
+  find_library(DATE_LIBRARY_RELEASE NAMES date-tz)
+  find_library(DATE_LIBRARY_DEBUG NAMES date-tzd)
+
+  include(SelectLibraryConfigurations)
+  select_library_configurations(DATE)
+endif()
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Date
+                                  REQUIRED_VARS DATE_LIBRARY DATE_INCLUDE_DIR
+                                  VERSION_VAR DATE_VERSION)
+
+if(DATE_FOUND)
+  set(DATE_INCLUDE_DIRS ${DATE_INCLUDE_DIR})
+  set(DATE_LIBRARIES ${DATE_LIBRARY})
+
+  if(ENABLE_INTERNAL_TZDATA)
+    set(DATE_DEFINITIONS "-DDATE_INTERNAL_TZDATA")
+  endif()
+
+  if(DATE_HAS_STRINGVIEW)
+    set(DATE_DEFINITIONS "-DDATE_HAS_STRINGVIEW")
+  endif()
```

These defines need to be appended, otherwise only the last one set will apply:
```suggestion
  if(ENABLE_INTERNAL_TZDATA)
    set(DATE_DEFINITIONS ${DATE_DEFINITIONS} -DDATE_INTERNAL_TZDATA)
  endif()

  if(DATE_HAS_STRINGVIEW)
    set(DATE_DEFINITIONS ${DATE_DEFINITIONS} -DDATE_HAS_STRINGVIEW)
  endif()
```

---

Yep, I think it is the reason of the crashes below - plus maybe the directory is copied in the wrong tree! Thanks for testing!